// Channel wrapper for sim engine to data broker messages.
//
// Rick Stovall, 2009-03-18
//
// This file released under the GPL 3.0 license.
// 
// Note: The message defined here is a "channel wrapper", a container for 
//  all messages between the simulation engine and the data broker.

package nrtb_msg;

import "ack_nak.proto";
import "sim_obj_tq_update.proto";
import "sim_management.proto";

message sim_to_db {
  required uint32 msg_uid = 1;    // the msg_uid must be unique for 
                                  //  for each message on a channel.
  // all the below messages are defined in 
  // the files included above.
  optional message_ack ack = 2;
  optional message_nak nak = 3;
  optional tq_data quanta_results = 4;
  optional sim_setup_data sim_setup = 5;
  optional sim_termination_data sim_end = 6;
  optional db_request_sim_stop db_abort = 7;
}
